api/src/main/scala/hmda/api/http/public/HmdaFilerPaths.scala

```diff
@@ -54,6 +61,47 @@ trait HmdaFilerPaths extends HmdaCustomDirectives with HmdaFilerProtocol with Ap
             sendResponse(filersF, uri)
           }
         }
+      } ~
+      path("filers" / Segment / Segment / "msaMds") { (period, instId) =>
+        encodeResponse {
+          timedGet { uri =>
+            val fSubmissions = (supervisor ? FindSubmissions(SubmissionPersistence.name, instId, period)).mapTo[ActorRef]
+            val hmdaFilerPersistenceF = (supervisor ? FindHmdaFilerPersistence(HmdaFilerPersistence.name)).mapTo[ActorRef]
+
+            val fMsas = for {
+              subActor <- fSubmissions
+              submissions <- (subActor ? GetState).mapTo[Seq[Submission]]
+
+              manager <- (supervisor ? FindProcessingActor(SubmissionManager.name, getLatestSignedSub(submissions))).mapTo[ActorRef]
+              larStats <- (manager ? GetActorRef(SubmissionLarStats.name)).mapTo[ActorRef]
+              stats <- (larStats ? FindIrsStats(getLatestSignedSub(submissions))).mapTo[Seq[Msa]]
+
+              a <- hmdaFilerPersistenceF
+              filer <- (a ? FindHmdaFiler(instId)).mapTo[Option[HmdaFiler]]
+            } yield (stats, filer)
+
+            onComplete(fMsas) {
+              case Success((msas, instOpt)) =>
+                val msasJson = msas.filterNot(_.id == "NA").map(msa => s"""{"id":"${msa.id}","name":"${msa.name}"}""").mkString("[", ",", "]")
+                val inst = instOpt.getOrElse(HmdaFiler("", "", "", ""))
+                if (inst.institutionId == "") {
+                  complete(ToResponseMarshallable(StatusCodes.NotFound -> s"Unable to find institution $instId"))
+                } else {
+                  val response =
+                    s"""
```

Comment:
What's the reasoning for manipulating raw JSON here instead of having a Scala object with a JSON protocol (similar to the `HmdaFiler` object above)?

Reply:
The `HmdaFiler` API has a separate actor/persistence model that stores and retrieves specific objects.  The Filer MSA endpoint aggregates information from three different sources, and so I opted for creating the JSON in place.